### DISCUSSION TITLE
branchの命名規則

### DISCUSSION THREAD

**Discussion by katataku** [katataku/minishell#12] [General]
命名規則どうします？

案：[issueチケット番号]/[シンプルな名前]
→例 0009/parse_token

**Comment by hayashi-ay** [comment]
feat/9_parse_token

**Reply by hayashi-ay** [reply]
https://qiita.com/c6tower/items/fe2aa4ecb78bef69928f

**Reply by hayashi-ay** [reply]
Issue番号を含めるのと簡単な名前をつけるのには賛成です。

**Reply by hayashi-ay** [reply]
あとはフォーマットとブランチprefixをつけるかどうかくらいですね〜。

**Reply by katataku** [reply]
おお、フォーマットいいですね！
ブランチの種類ってmaster/feature/hotfixぐらいでしょうか。

**Reply by hayashi-ay** [reply]
今回だとfeature/fixくらいですかね〜？

**Reply by hayashi-ay** [reply]
実運用しているサービスとかだとブランチフローを無視してhotfixとかも使ったりしますが。

**Reply by katataku** [reply]
fix:developブランチの中の種類の一つ。featureと並列
hotfix:ブランチの種類の一つ。developと並列。

上の理解であってますか？
であれば、fixが正しかったですね＞＜

**Reply by hayashi-ay** [reply]
ブランチ戦略の話ですね〜。
https://qiita.com/tatane616/items/aec00cdc1b659761cf88

商用を想定しているサービスとかだと、
開発用にdevelop, 商用にmainブランチを用意していて、
Issue単位でfeatブランチを切ってそこで開発して、developにマージ。リリースの単位でmainブランチにマージみたいなフローをしたりします。
hotfixはfeat->develop->mainの流れを無視して、緊急の不具合対応とかのときにhotfix->mainのように直接mainにマージするブランチの命名です。

**Reply by hayashi-ay** [reply]
Issueの種別（機能追加or既存機能の不具合修正）でfeat or fixでブランチ名のprefixを分けたりもします。実際やっていて結構線引が曖昧なので区別しなくてfeatで統一してるところもあるかも？

**Reply by katataku** [reply]
なるほど。色々間違ってました＞＜

今回のminishellだと、GitHub-Flowをベースに、featuerブランチが「Issueの種別（機能追加or既存機能の不具合修正）でfeat or fixでブランチ名のprefixを分けたり」する感じですかね？

**Reply by hayashi-ay** [reply]
そんなイメージでしました。
mainブランチがあってそこからIssueの種別に応じてfeat/fixブランチを切って問題なければmainにマージくらいがちょうどよいかなと思ってます。

**Comment by hayashi-ay** [comment]
Issue番号の0埋めだけ反対です〜。めんどくさいので。
